src/script/team/TeamFeatureUpdater/TeamFeatureUpdater.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {FeatureList, FEATURE_KEY, FeatureStatus} from '@wireapp/api-client/lib/team';
+
+export enum FeatureUpdateType {
+  ENABLED = 'ENABLED', // Feature was enabled or didn't exist before and now it was added and is enabled
+  DISABLED = 'DISABLED', // Feature was previously enabled and now it was removed or disabled
+  CONFIG_CHANGED = 'CONFIG_CHANGED', // Feature was enabled and now it's config has changed
+  UNCHANGED = 'UNCHANGED', // Feature is enabled or disabled or doesn't exist and it's config hasn't changed
+}
+
+type FeatureUpdateEnabled<Key extends FEATURE_KEY> = {
+  type: FeatureUpdateType.ENABLED;
+  prev?: FeatureList[Key];
+  next: FeatureList[Key];
+};
+
+type FeatureUpdateDisabled<Key extends FEATURE_KEY> = {
+  type: FeatureUpdateType.DISABLED;
+  prev: FeatureList[Key];
+  next?: FeatureList[Key];
+};
+
+type FeatureUpdateConfigChanged<Key extends FEATURE_KEY> = {
+  type: FeatureUpdateType.CONFIG_CHANGED;
+  prev: FeatureList[Key];
+  next: NonNullable<FeatureList[Key]>;
+};
+
+type FeatureUpdateUnchanged<Key extends FEATURE_KEY> = {
+  type: FeatureUpdateType.UNCHANGED;
+  prev?: FeatureList[Key];
+  next?: FeatureList[Key];
+};
+
+type FeatureUpdate<Key extends FEATURE_KEY> =
+  | FeatureUpdateEnabled<Key>
+  | FeatureUpdateDisabled<Key>
+  | FeatureUpdateConfigChanged<Key>
+  | FeatureUpdateUnchanged<Key>;
+
+export const getTeamFeatureUpdate = <Key extends FEATURE_KEY>(
```

Comment:
That pretty clean and neat API 👏 
Love it

Reply:
Addressed with [287ee04](https://github.com/wireapp/wire-webapp/pull/16675/commits/287ee04de9652be1da5aee66310d516dff6919c3) 😉